src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -109,19 +119,30 @@
     @FXML
     private FooterController footerController;
 
-    /**
-     * Indicate whether the user's hard drive is running out of space for game downloads.
-     */
-    private final Property<Optional<Warning>> warning;
-    private Property<Tooltip> playButtonTooltip;
-    private Property<Package> selectedPackageProperty;
-
     public ApplicationController() {
         warning = new SimpleObjectProperty<>(Optional.empty());
-        selectedPackageProperty = new SimpleObjectProperty<>(null);
         gameService = new GameService();
         gameService.setOnFailed(this::handleRunFailed);
         gameService.valueProperty().addListener(this::handleRunStarted);
+
+        downloading = new SimpleBooleanProperty(false);
+
+        selectedRelease = new SimpleObjectProperty<>();
+
+        installedGames = FXCollections.observableSet();
+
+        gameAction = new SimpleObjectProperty<>(GameAction.DOWNLOAD);
+        gameAction.bind(Bindings.createObjectBinding(() -> {
+            final GameRelease release = selectedRelease.getValue();
+            final boolean isInstalled = release != null && installedGames.contains(release.getId());
```

Comment:
```
            final boolean isInstalled = release != null && ...
```
Am I the only one that needs to read this twice to parse? :see_no_evil:

Reply:
I don't know - this was suggested by IntelliJ, and I think it kind of makes sense :monocle_face: 

I wanted to write
```java
final boolean isInstalled = installedGames.contains(release.getId()); 
```
but that may fail if `release` is not defined. So, instead I naively wrote
```java
final boolean isInstalled = release == null ? false : installedGames.contains(release.getId()); 
```
which is basically the same as "_guarding_" the access with `release != null &&`. I think that's also a quite common pattern in other languages :shrug:

Reply:
maybe I'm just not used to it... reads quite weird in my eyes, but if that's a common pattern I guess it's okay